File: test/language/templates/python/SubtypeTemplateArgumentTest.py
import Templates


class SubtypeTemplateArgumentTest(Templates.TestCase):
    def testBitSizeOf(self):
        field_uint32 = self.api.Field_uint32(10)
        field_compound = self.api.Field_Compound(self.api.Compound(10))
        subtypeTemplateArgument = self.api.SubtypeTemplateArgument(
            field_uint32, field_uint32, field_uint32, field_compound, field_compound, field_compound
        )
        self.assertEqual(192, subtypeTemplateArgument.bitsizeof())

File: test/language/templates/cpp/SubtypeTemplateArgumentTest.cpp
#include "gtest/gtest.h"
#include "templates/subtype_template_argument/SubtypeTemplateArgument.h"

namespace templates
{
namespace subtype_template_argument
{

TEST(SubtypeTemplateArgumentTest, bitSizeOf)
{
    const Field_uint32 field_uint32(10);
    const Field_Compound field_compound(Compound(10));
    const SubtypeTemplateArgument subtypeTemplateArgument(
            field_uint32, field_uint32, field_uint32, field_compound, field_compound, field_compound);
    ASSERT_EQ(192, subtypeTemplateArgument.bitSizeOf());
}

} // namespace subtype_template_argument
} // namespace templates
